### PR TITLE
fix: make compiler happy

### PR DESCRIPTION
**Pull request by thofma**

### CODE REVIEW

File: src/generic/MPoly.jl [hunk 1]
```diff
@@ -66,7 +66,7 @@ ring.
 """
 function gens(a::MPolyRing{T}) where {T <: RingElement}
    n = a.num_vars
-   return [gen(a, i, Val{a.ord}) for i in 1:n]
+   return elem_type(a)[gen(a, i, Val{a.ord})::elem_type(a) for i in 1:n]
```
Comment (lgoettgens) [inline src/generic/MPoly.jl:69]:
```suggestion
   return elem_type(a)[gen(a, i, Val(a.ord))::elem_type(a) for i in 1:n]
```
`Val` objects are supposed to be created with `()`, the `{}` is their type

Comment (thofma) [reply src/generic/MPoly.jl:69]: Yes, but the methods expect their type. It is unstable anyway.

Comment (lgoettgens) [reply src/generic/MPoly.jl:69]: Ok. I don't really care.

**Comment by lgoettgens** [comment]
If re-running the in validations job enough times makes it green, this is fine from my pov

File: src/generic/MPoly.jl (hunk 1 above)
Comment (thofma) [reply src/generic/MPoly.jl:69]: `Val{a.ord}` and `Val(a.ord)` do different things and unless the suggestion was to rewrite everyhing, I don't understand the comment.

Comment (lgoettgens) [reply src/generic/MPoly.jl:69]: please just ignore the comment

**Pull request merged**